### PR TITLE
Fix test-capability

### LINKED ISSUE #10663: test-capability seems to be failing in docker containers
This was originally discovered by @mrc0mmand in https://github.com/systemd/systemd/pull/10644:
```
The output from the failed tests:
298/499 test-capability                         FAIL     0.02 s (killed by signal 6 SIGABRT)
--- command ---
SYSTEMD_KBD_MODEL_MAP='/build/src/locale/kbd-model-map' PATH='/build/build:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin' SYSTEMD_LANGUAGE_FALLBACK_MAP='/build/src/locale/language-fallback-map' /build/build/test-capability
--- stderr ---
have ambient caps: yes
Capabilities:= cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read+eip
Capabilities:= cap_dac_override,cap_net_raw+ep
Capabilities:= cap_dac_override+ep
Failed to drop auxiliary groups list: Operation not permitted
Failed to change group ID: Operation not permitted
Assertion 'fv == CAP_CLEAR' failed at ../src/test/test-capability.c:184, function test_update_inherited_set(). Aborting.
-------
```

The output of `strace` can be found at https://paste.fedoraproject.org/paste/-UajhkU~M9F-NyOZyS28vA.

I'm adding the "priority" label because TravisCI is broken due to this issue. I temporarily added a kludge that overwrites `test-capability` with a shell scripts exiting with 77.

### PR DESCRIPTION
**Pull request by mrc0mmand**
Docker's default capability set has the inherited flag already
set - that breaks tests which expect otherwise. Let's just
drop the check and run the test anyway.

Fixes #10663

```
# build/test-capability
have ambient caps: yes
Capabilities:= cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read+eip
Capabilities:= cap_dac_override,cap_net_raw+ep
Capabilities:= cap_dac_override+ep
Failed to drop auxiliary groups list: Operation not permitted
Failed to change group ID: Operation not permitted
test_update_inherited_set: Inheritable flag is already set, skipping test
test_set_ambient_caps: Inheritable flag is already set, skipping test
# echo $?
0
```

### CODE REVIEW

File: src/test/test-capability.c [hunk 1]
```diff
@@ -181,6 +181,12 @@ static void test_update_inherited_set(void) {
         caps = cap_get_proc();
         assert_se(caps);
         assert_se(!cap_get_flag(caps, CAP_CHOWN, CAP_INHERITABLE, &fv));
+
+        if (fv == CAP_SET) {
+            log_info("%s: Inheritable flag is already set, skipping test", __FUNCTION__);
+            return;
```
Comment (yuwata) [inline src/test/test-capability.c]: Please use 8ch indentation...

Comment (mrc0mmand) [reply src/test/test-capability.c]: I knew I forgot something, fixed, thanks!

File: src/test/test-capability.c [hunk 2]
```diff
@@ -181,6 +181,12 @@ static void test_update_inherited_set(void) {
         caps = cap_get_proc();
         assert_se(caps);
         assert_se(!cap_get_flag(caps, CAP_CHOWN, CAP_INHERITABLE, &fv));
+
+        if (fv == CAP_SET) {
+                log_info("%s: Inheritable flag is already set, skipping test", __FUNCTION__);
+                return;
+        }
```
Comment (poettering) [inline src/test/test-capability.c]: let's just drop the precondition check alltogether, if it's not universally true. i.e. drop the cap_get_flag() assert above, and the == CAP_CLEAR assert below.

if the preconditions don't hold, let's not do them. We can still run the rest of the test and verify that the effect is systematics regardless what the preconditions are...

File: src/test/test-capability.c [hunk 3]
```diff
@@ -200,6 +206,12 @@ static void test_set_ambient_caps(void) {
...
+
         assert(fv == CAP_CLEAR);
```
Comment (poettering) [inline src/test/test-capability.c]: as above, let's rather drop the precondition, but still run the test

File: src/test/test-capability.c (hunk 2 above)
Comment (mrc0mmand) [reply src/test/test-capability.c]: makes sense, fixed

**Pull request merged**